src/workarounds/forwarddiff_rules.jl

```diff
@@ -236,7 +236,10 @@ function self_consistent_field(basis_dual::PlaneWaveBasis{T};
     # TODO Could add δresults[α].δVind the dual part of the total local potential in ham_dual
     # and in this way return a ham that represents also the total change in Hamiltonian
 
+    # TODO differentiate occupation = compute_occupation(...)
+
     merge(scfres, (; ψ, ρ, eigenvalues, basis=basis_dual,
+                   energies=energies_dual, ham=ham_dual,
```

Comment:
TODO remove the merge, and put all the fields in explicitly

Reply:
done in [c17293b](https://github.com/JuliaMolSim/DFTK.jl/pull/709/commits/c17293bb8035210caf358a1d35419a6093563bdf)